framework-plugins/lisk-framework-http-api-plugin/src/controllers/transactions.ts

```diff
@@ -109,24 +109,21 @@ export const postTransaction = (channel: BaseChannel, codec: PluginCodec) => asy
 		return;
 	}
 
-	const encodedTransaction = codec.encodeTransaction(req.body as TransactionInput);
-
-	const result = await channel.invoke<{
-		transactionId?: string;
-		message?: string;
-		errors?: Error[] | Error;
-	}>('app:postTransaction', {
-		transaction: encodedTransaction,
-	});
+	try {
+		const encodedTransaction = codec.encodeTransaction(req.body as TransactionInput);
 
-	if (result.errors) {
-		const processingErrors = (result.errors as Error[]).map(e => ({
-			message: e.message,
-		}));
+		const result = await channel.invoke<{
+			transactionId?: string;
+			message?: string;
+			errors?: Error[] | Error;
+		}>('app:postTransaction', {
+			transaction: encodedTransaction,
+		});
 
-		res.status(409).json({ errors: processingErrors });
-		return;
+		res.status(200).json({ data: result, meta: {} });
+	} catch (err) {
+		res
+			.status(409)
+			.json({ error: (err as Error).message.replace('Lisk validator found 1 error[s]:\n', '') });
```

Comment:
I think format should be `{ errors: [] }`, and why is it replacing?

Reply:
As the method being invoked returns errors using lisk-validator errors it seems this method returns only an error and with the lisk validator message before the actual error. Maybe we should create a different error handler logic for the action?

Reply:
I updated the format. Kept the replace as in some cases as the action `handleEventPostTransaction` is using `LiskValidationError` maybe we should consider add an extra error handler different to this that just process and returns the errors with no additional text?